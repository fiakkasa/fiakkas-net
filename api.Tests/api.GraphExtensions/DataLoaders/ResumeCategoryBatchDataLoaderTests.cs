using api.Categories.Enums;
using api.Categories.Interfaces;
using api.Categories.Models;

namespace api.GraphExtensions.DataLoaders.Tests;

public class ResumeCategoryBatchDataLoaderTests
{
    [Fact]
    public async Task LoadBatchAsync_Should_Return_Data_When_Matches_Found()
    {
        var dataRepository = new MockDataRepository<ICategoryEntity>(
        [
            new CategoryEntity
            {
                Kind = CategoryType.Resume,
                Id = new Guid("eb9d6258-99c4-46bd-bd44-23d35b19965d"),
                CreatedAt = new(2024, 1, 1, 0, 0, 0, TimeSpan.Zero),
                UpdatedAt = null,
                Version = 1,
                Title = "Title"
            }
        ]);

        var sut = new ResumeCategoryBatchDataLoader(dataRepository, AutoBatchScheduler.Default);

        var result = await sut.LoadAsync([new Guid("eb9d6258-99c4-46bd-bd44-23d35b19965d")], CancellationToken.None);

        result.Should().NotBeEmpty();
        result.MatchSnapshot();
    }

    [Fact]
    public async Task LoadBatchAsync_Should_Return_Empty_Collection_When_No_Matches_Found()
    {
        var dataRepository = new MockDataRepository<ICategoryEntity>();

        var sut = new ResumeCategoryBatchDataLoader(dataRepository, AutoBatchScheduler.Default);

        var result = await sut.LoadAsync([Guid.NewGuid()], CancellationToken.None);

        result.Should().NotBeEmpty();
        result.MatchSnapshot();
    }
}
