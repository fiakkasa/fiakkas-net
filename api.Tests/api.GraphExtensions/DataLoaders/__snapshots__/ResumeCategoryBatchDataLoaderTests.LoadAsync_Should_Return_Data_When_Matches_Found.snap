[
  {
    "AssociatedCategoryTypes": [],
    "Title": "Title",
    "Id": "eb9d6258-99c4-46bd-bd44-23d35b19965d",
    "CreatedAt": "2024-01-01T00:00:00+00:00",
    "UpdatedAt": null,
    "Version": 1
  }
]
